pkg/manager/manager.go

```
@@ -670,8 +773,94 @@ func UpgradeRule(cursorDir string, ruleKey string) error {
+		matches := githubBlobPattern.FindStringSubmatch(ruleToUpgrade.Reference)
+		if len(matches) < 5 {
+			return fmt.Errorf("invalid GitHub blob URL: %s", ruleToUpgrade.Reference)
+		}
+
+		owner := matches[1]
+		repo := matches[2]
+		gitRef := matches[3]
+		_ = matches[4] // Path is not used in this context
+
+		// If we have a branch name, check if there's a new commit
+		if !isGitCommitHash(gitRef) {
```

golangci-lint (1.64.8)

`if !isGitCommitHash(gitRef)` has complex nested blocks (complexity: 20)

(nestif)